qiskit/algorithms/eigensolvers/vqd.py

```diff
@@ -0,0 +1,512 @@
+    such as when there are reasons to believe that the solution point is close to a particular
+    point.
+
+    The length of the *initial_point* list value must match the number of the parameters
+    expected by the ansatz being used. If the *initial_point* is left at the default
+    of ``None``, then VQD will look to the ansatz for a preferred value, based on its
+    given initial state. If the ansatz returns ``None``,
+    then a random point will be generated within the parameter bounds set, as per above.
+    If the ansatz provides ``None`` as the lower bound, then VQD
+    will default it to :math:`-2\pi`; similarly, if the ansatz returns ``None``
+    as the upper bound, the default value will be :math:`2\pi`.
+
+    """
+
+    def __init__(
+        self,
+        estimator: BaseEstimator,
+        fidelity: BaseStateFidelity,
+        ansatz: QuantumCircuit | None = None,
+        *,
+        k: int = 2,
+        betas: list[float] | None = None,
+        optimizer: Optimizer | Minimizer | None = None,
+        initial_point: np.ndarray | None = None,
+        gradient: Callable | None = None,
+        callback: Callable[[int, np.ndarray, float, float], None] | None = None,
+    ) -> None:
+        """
+
+        Args:
+            estimator: The estimator primitive.
+            fidelity: The fidelity class using primitives.
+            ansatz: A parameterized circuit used as ansatz for the wave function.
+            k: the number of eigenvalues to return. Returns the lowest k eigenvalues.
+            betas: beta parameters in the VQD paper.
+                Should have length k - 1, with k the number of excited states.
+                These hyperparameters balance the contribution of each overlap term to the cost
+                function and have a default value computed as the mean square sum of the
+                coefficients of the observable.
+            optimizer: A classical optimizer. Can either be a Qiskit optimizer or a callable
+                that takes an array as input and returns a Qiskit or SciPy optimization result.
+            initial_point: An optional initial point (i.e. initial parameter values)
+                for the optimizer. If ``None`` then VQD will look to the ansatz for a preferred
+                point and if not will simply compute a random one.
+            gradient: An optional fidelity gradient using primitives.
+            callback: a callback that can access the intermediate data during the optimization.
+                Four parameter values are passed to the callback as follows during each evaluation
+                by the optimizer for its current set of parameters as it works towards the minimum.
+                These are: the evaluation count, the optimizer parameters for the ansatz, the
+                evaluated mean, the evaluated standard deviation, and the current step.
+        """
+        super().__init__()
+
+        self.estimator = estimator
```

Comment:
In going over to public instance vars instead of getter/setters we need to document that these are available/part of the public interface which can be done using Attributes. See the PVQD algo/PR if you need an example.